auth/src/main/java/com/firebase/ui/auth/ui/email/RegisterEmailActivity.java

```diff
@@ -143,13 +289,13 @@ public void onComplete(@NonNull Task<Void> task) {
                                         // the credential to SmartLock (if enabled).
                                         mActivityHelper.saveCredentialsOrFinish(
                                                 mSaveSmartLock,
-                                                firebaseUser,
+                                                user,
                                                 password);
                                     }
                                 });
                     }
                 })
-                .addOnFailureListener(new OnFailureListener() {
+                .addOnFailureListener(this, new OnFailureListener() {
```

Comment:
Any reason we're only tying the failure listener to the Activity and not the success listener?